cilium/cmd/service_update.go

```diff
@@ -50,6 +52,8 @@ func init() {
 	serviceUpdateCmd.Flags().BoolVarP(&k8sClusterInternal, "k8s-cluster-internal", "", false, "Set service as cluster-internal for externalTrafficPolicy=Local")
 	serviceUpdateCmd.Flags().StringVarP(&frontend, "frontend", "", "", "Frontend address")
 	serviceUpdateCmd.Flags().StringSliceVarP(&backends, "backends", "", []string{}, "Backend address or addresses (<IP:Port>)")
+	serviceUpdateCmd.Flags().StringSliceVarP(&backendStates, "states", "", []string{}, "Backend state(s) as {active(default),terminating,quarantined,maintenance}")
+	serviceUpdateCmd.Flags().BoolVarP(&updateServices, "update-services", "", false, "Update all services selecting the backends with their given states (id and frontend are ignored)")
```

Comment:
while I understand why this flag is added, it's a little confusing to me :(.

Just curious if you have considered to have a sub command (e.g. cilium service update state --backends 1.1.1.1:80 --stage quarantined)

Also, is there any case that --states flag can be used independently with --update-services flag ?

Reply:
>Also, is there any case that --states flag can be used independently with --update-services flag ?

There is no use case right now. 

Regarding sub-command, we internally discussed to add a new flag.

Reply:
Right now you have `cilium service update --backends 10.244.0.58:80 --states quarantined --updates-services=true`.

The last `--updates-services=true` is very odd from a user pov, why can't we imply this when no frontend parameter was passed so that a `cilium service update --backends 10.244.0.58:80 --states quarantined` is sufficient?

Reply:
@borkmann I had brought this up last month, and we decided to not rely on implicit assumptions, rather make the API more clear by adding a flag. I'll cc you to that thread.

Reply:
Removed the flag as discussed offline.